eq-author/src/App/page/Logic/Routing/RoutingPage/NoRouting.test.js

```diff
@@ -1,26 +1,211 @@
 
 describe("components/NoRouting", () => {
-  it("should render", () => {
-    const wrapper = shallow(
-      <NoRouting onAddRouting={jest.fn()} title="Test">
-        Ullamcorper Venenatis Fringilla
-      </NoRouting>
+  const disabledTitle = "Routing is not available for this quesiton";
+  const disabledParagraph =
+    "You can't route on the last question in a questionnaire.";
+  const enabledTitle = "No routing rules exist for this question";
+  const enabledParagraph =
+    "Users completing this question will be taken to the next page.";
+
+  it("should render with button enabled", () => {
+    let props, questionnaire, page;
+
+    questionnaire = {
+      sections: [
+        {
+          id: 1,
+          title: "Section-1",
+          position: 0,
+          pages: [
+            {
+              id: "page-1",
+              pageType: "QuestionPage",
+              title: "<p>Questions 1</p>",
+              position: 0,
+              description: "",
+              answers: [
+                {
+                  id: "ans-p1-1",
+                  description: "",
+                  guidance: "",
+                  label: "num1",
+                  questionPageId: "qp-1",
+                  secondaryLabel: null,
+                },
+              ],
+            },
+            {
+              id: "page-2",
+              pageType: "QuestionPage",
+              title: "<p>Questions 2</p>",
+              position: 1,
+              description: "",
+              answers: [
+                {
+                  id: "ans-p2-1",
+                  description: "",
+                  guidance: "",
+                  label: "num2",
+                  questionPageId: "qp-2",
+
+                  secondaryLabel: null,
+                },
+              ],
+            },
+          ],
+        },
+      ],
+    };
+
+    page = {
+      id: "page-1",
+      pageType: "QuestionPage",
+      title: "<p>Questions 1</p>",
+      position: 0,
+      description: "",
+      answers: [
+        {
+          id: "ans-p1-1",
+          description: "",
+          guidance: "",
+          label: "num1",
+          questionPageId: "qp-1",
+          secondaryLabel: null,
+        },
+      ],
+    };
+
+    props = {
+      questionnaire,
+      page,
+    };
+    
+    const { queryByText } = render(
+      <QuestionnaireContext.Provider value={{ props }}>
+        <NoRouting onAddRouting={jest.fn()} title="Test" page={page} questionnaire={questionnaire}>
+          Ullamcorper Venenatis Fringilla
+        </NoRouting>
+      </QuestionnaireContext.Provider>
+    );
+
+    expect(queryByText(enabledTitle)).toBeTruthy();
+    expect(queryByText(enabledParagraph)).toBeTruthy();
+    expect(queryByText(disabledTitle)).toBeFalsy();
+    expect(queryByText(disabledParagraph)).toBeFalsy();
+
+    expect(screen.getByTestId("btn-add-routing")).not.toHaveStyle(
+      `pointer-events: none; 
+       opacity: 0.6;`
+    );
+  });
```

Comment:
Does this check to see if the button isn't disabled? If so, you can use this: `.toHaveAttribute('disabled')` and it'll live longer. If we ever change the button styles, which might happen if their contrast becomes an issue for accessibility, then this assertion would fail if the opacity changes.

Reply:
Added disabled={isLastPage} into AddRoutingButton which prevents keyboard focus too

Reply:
Awesome. Cheers Josh. In that case, could you update this test to assert on that instead please?